### DISCUSSION TITLE
org.openqa.selenium.SessionNotCreatedException: Could not start a new session. Response code 500. Message: An unknown server-side error occurred while processing the command. Original error: Failed to set language: en and country: EN

### DISCUSSION THREAD

**Discussion by devikabontalakoti** [appium/appium#19696] [Q&A]
Build info: version: '4.13.0', revision: 'ba948ece5b*'
System info: os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '13.4', java.version: '17.0.6'
Driver info: io.appium.java_client.android.AndroidDriver
Command: [null, newSession {capabilities=[{appium:appActivity=com.zodiac.fluidrapro.MainActivity, appium:appPackage=com.zodiac.fluidrapro, appium:autoDismissAlerts=true, appium:automationName=UiAutomator2, appium:deviceName=26171JEGR06223, appium:language=en, appium:locale=EN, platformName=ANDROID, appium:platformVersion=14, appium:unicodeKeyboard=true}], desiredCapabilities=Capabilities {appActivity: com.zodiac.fluidrapro.MainA..., appPackage: com.zodiac.fluidrapro, autoDismissAlerts: true, automationName: UiAutomator2, deviceName: 26171JEGR06223, language: en, locale: EN, platformName: ANDROID, platformVersion: 14, unicodeKeyboard: true}}]
Capabilities {appActivity: com.zodiac.fluidrapro.MainA..., appPackage: com.zodiac.fluidrapro, autoDismissAlerts: true, automationName: UiAutomator2, deviceName: 26171JEGR06223, language: en, locale: EN, platformName: ANDROID, platformVersion: 14, unicodeKeyboard: true}

Need your inputs to resolve this issue.

Thank you !!

**Comment by portyu9** [comment]
Hi @devikabontalakoti 

The problem is:

```text
appium:language = en
appium:locale   = EN
```

For UiAutomator2, `language` is the language code, while `locale` should be the matching country code.

For English (United States), use:

```java
UiAutomator2Options options = new UiAutomator2Options()
    .setLanguage("en")
    .setLocale("US");
```

For British English, for example:

```text
language = en
locale   = GB
```

So change `locale=EN` to a valid country/locale combination such as `US`, or remove the language/locale capabilities entirely if the test does not need to change the device locale.

Please mark this answer as accepted if it helped, thank you.
